### PR TITLE
[US-4563] Demoshop experiences documentation

### PR DESCRIPTION
**Pull request by simara-svatopluk**
| Q             | A
| ------------- | ---
|Description, reason for the PR| US-4563
|New feature| No
|BC breaks| No
|Fixes issues| 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docs/introduction/framework-extensibility.md [hunk 1]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
 * [Extending the application using standard Symfony techniques](https://symfony.com/doc/current/bundles/override.html)
     * e.g. overriding Twig templates, routes, services, ...
 * [Adding a new advert position](/docs/cookbook/adding-a-new-advert-position.md) to be used in the administration section *Marketing > Advertising system*
 * open-box modifications in `project-base`
     * e.g. adding new entities, changing the FE design, customization of FE javascripts, adding new FE pages (routes and controllers), ...
+* [Hiding the existing features and functionality](https://github.com/shopsys/demoshop/pull/13)
+
+## What is achievable with additional effort
+* [Extending factories and controllers](https://github.com/shopsys/demoshop/commit/898d111879aef40196f79ac763373560f44aef59#diff-1b3bd68670cd376165cdc6cfc634f24f)
+* Adding custom javascript into administration
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: should not be here some additional info or link? :slightly_smiling_face:

File: docs/introduction/framework-extensibility.md [hunk 2]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+* [Adding form option into existing form](https://github.com/shopsys/demoshop/commit/898d111879aef40196f79ac763373560f44aef59#diff-3293b000b06ad6c0280341584c4d661d)
+* [Extending administration theme](https://github.com/shopsys/demoshop/commit/d0e0eaaa2eeac5e1c90d8a29be5c827c4a067b9f)
+* [Changing an entity association](https://github.com/shopsys/demoshop/commit/9931083ea37ad611568e32bc1a9c8cf203401809)
+    * this change is complicated and potentially dangerous
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: Is it really?

File: docs/introduction/framework-extensibility.md [hunk 3]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
 * [Adding a new advert position](/docs/cookbook/adding-a-new-advert-position.md) to be used in the administration section *Marketing > Advertising system*
 * open-box modifications in `project-base`
     * e.g. adding new entities, changing the FE design, customization of FE javascripts, adding new FE pages (routes and controllers), ...
+* [Hiding the existing features and functionality](https://github.com/shopsys/demoshop/pull/13)
+
+## What is achievable with additional effort
+* [Extending factories and controllers](https://github.com/shopsys/demoshop/commit/898d111879aef40196f79ac763373560f44aef59#diff-1b3bd68670cd376165cdc6cfc634f24f)
+* Adding custom javascript into administration
+* [Adding form option into existing form](https://github.com/shopsys/demoshop/commit/898d111879aef40196f79ac763373560f44aef59#diff-3293b000b06ad6c0280341584c4d661d)
+* [Extending administration theme](https://github.com/shopsys/demoshop/commit/d0e0eaaa2eeac5e1c90d8a29be5c827c4a067b9f)
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: I do not like this - the link leads to the commit in the framework that enables the feature. However, some other points lead to demonstrations of such functionalities (e.g. Extending factories and controllers) so I am confused a little bit

File: docs/introduction/framework-extensibility.md [hunk 4]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+* [Changing an entity association](https://github.com/shopsys/demoshop/commit/9931083ea37ad611568e32bc1a9c8cf203401809)
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: again, should not this link rather to the commit where the association is overridden actually?

File: docs/introduction/framework-extensibility.md [hunk 5]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+    * this change is complicated and potentially dangerous
 
 ## Which issues are going to be addressed soon
 * Extending data fixtures (including performance data fixtures)
 * Extending data grids in the administration
 * Extending all forms in the administration without the need of the template overriding
-* Hiding the existing features and functionality
+* Extending classes like Repositories don't lead to change of project-base tests
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: I would rewrite slightly to "Extending classes like Repositories without the need for changing the project-base tests"

File: docs/introduction/framework-extensibility.md [hunk 6]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+* Incorrectly extended entities cause clear exceptions
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: I would remove this line, it is confusing without a context

File: docs/introduction/framework-extensibility.md [hunk 7]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
 
 ## What is not achievable
 * Removing an attribute from a framework entity
-* Changing a data type of an entity attribute or association
-* Removing existing entities and features
\ No newline at end of file
+* Changing a data type of an entity attribute
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: :thinking: when changing a data type of association is possible, should not it be possible for entity attributes as well? :slightly_smiling_face:

File: docs/introduction/framework-extensibility.md [hunk 8]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+    * pick up places are downloaded by cron
+    * order process change
+    * details in a [issue description](https://github.com/shopsys/demoshop/issues/3)
+* [Product attribute "condition"](https://github.com/shopsys/demoshop/pull/7)
+    * product entity extension
+    * administration form extension
+    * frontend product change
+    * google feed change
+    * detailed info in a [issue description](https://github.com/shopsys/demoshop/issues/4)
+* [Category second description](https://github.com/shopsys/demoshop/pull/8)
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: rather "Second description of a category"?

File: docs/introduction/framework-extensibility.md [hunk 9]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+    * category entity extension
+    * administration form extension
+        * new domain attribute
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: rather "multidomain"

File: docs/introduction/framework-extensibility.md [hunk 10]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+    * frontend product list change
+    * detailed info in a [issue description](https://github.com/shopsys/demoshop/issues/5)
+* [Twig templates cache](https://github.com/shopsys/demoshop/pull/9)
+    * performance improved by ~15%
+    * cache invalidation is done by 5 minutes interval
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: "cache is invalidated every 5 minutes"?

File: docs/introduction/framework-extensibility.md [hunk 11]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+* [Hidden the functionality of the flags](https://github.com/shopsys/demoshop/pull/13)
+    * hidden functionality in administration
+    * hidden functionality in frontend
+    * flags do not affect shop at all
+* [Company account with multiple users](https://github.com/shopsys/demoshop/pull/15)
+    * group user accounts under one company account
+    * separate users login credentials
+    * share users' attributes
+    * change association from 1:1 to 1:N
+        * this change is complicated and potentially dangerous, but achievable
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: Do we have a reason why to call this "complicated and potentially dangerous"? It sounds weird to me...

File: docs/introduction/framework-extensibility.md [hunk 12]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
+    * performance improved by ~15%
+    * cache invalidation is done by 5 minutes interval
+* [Hidden the functionality of the flags](https://github.com/shopsys/demoshop/pull/13)
+    * hidden functionality in administration
+    * hidden functionality in frontend
+    * flags do not affect shop at all
+* [Company account with multiple users](https://github.com/shopsys/demoshop/pull/15)
+    * group user accounts under one company account
+    * separate users login credentials
+    * share users' attributes
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: "share company attributes"?

File: docs/wip_glassbox/entity-extension.md [hunk 13]
```diff
@@ -138,5 +138,7 @@ DiscriminatorMap must always contain descendants' FQN because LoadORMMetadataSub
   * in hydrated Doctrine references
   * in the EntityManager, Repositories and QueryBuilders
   * in newly created entities
+* If You are running tests, update also `\Tests\ShopBundle\Database\EntityExtension\EntityExtensionTest`
```
Comment (vitek-rostislav) [inline docs/wip_glassbox/entity-extension.md]: I would rewrite to simpler "If You are running tests, update also `\Tests\ShopBundle\Database\EntityExtension\EntityExtensionTest` - add your extended entity into `$entityExtensionMap` in the `setUp()` method"

File: docs/introduction/framework-extensibility.md [hunk 14]
```diff
@@ -13,22 +13,64 @@ as well as a list of customizations that are not (and will not be) possible at a
...
 * [Implementing custom product feed or modifying an existing one](/docs/introduction/product-feeds.md)
 * [Extending the application using standard Symfony techniques](https://symfony.com/doc/current/bundles/override.html)
     * e.g. overriding Twig templates, routes, services, ...
 * [Adding a new advert position](/docs/cookbook/adding-a-new-advert-position.md) to be used in the administration section *Marketing > Advertising system*
 * open-box modifications in `project-base`
     * e.g. adding new entities, changing the FE design, customization of FE javascripts, adding new FE pages (routes and controllers), ...
+* [Hiding the existing features and functionality](https://github.com/shopsys/demoshop/pull/13)
+
+## What is achievable with additional effort
+* [Extending factories and controllers](https://github.com/shopsys/demoshop/commit/898d111879aef40196f79ac763373560f44aef59#diff-1b3bd68670cd376165cdc6cfc634f24f)
```
Comment (vitek-rostislav) [inline docs/introduction/framework-extensibility.md]: In this section, I would probably explain in 1 sentence what does the "additional effort" mean. When I think about that, should not this be rather moved into "Which issues are going to be addressed soon" section? "Extending admin controllers and some factory classes is difficult as their properties and methods have private visibility"

File: docs/introduction/framework-extensibility.md (hunk 1 above)
Comment (simara-svatopluk) [reply docs/introduction/framework-extensibility.md]: I didn't found the resource for this, so I'm removing this line

File: docs/introduction/framework-extensibility.md (hunk 3 above)
Comment (simara-svatopluk) [reply docs/introduction/framework-extensibility.md]: I don't understand, it links to demoshop

**Pull request merged**